src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -140,6 +141,14 @@ public Instance(
             this.refreshBeansPeriod = appConfig.getRefreshBeansPeriod();
         }
 
+        this.service = (String) instanceMap.get("service");
+        if ((this.service == null || this.service == "") && initConfig != null) {
```

Comment:
I think it'll be better to do:
```java
        if ((this.service == null || this.service.isEmpty()) && initConfig != null) {
```
You may already know but you should not use `==` with string in Java, if you don't want to go with `.isEmpty` you should use `.equals`